src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -67,13 +69,72 @@ impl ops::AddAssign for ItemCount {
     }
 }
 
-#[derive(Default)]
 struct CoverageCalculator {
     items: BTreeMap<FileName, ItemCount>,
+    output_format: Option<OutputFormat>,
+}
+
+fn limit_filename_len(filename: String) -> String {
+    // if a filename is too long, shorten it so we don't blow out the table
+    // FIXME(misdreavus): this needs to count graphemes, and probably also track
+    // double-wide characters...
+    if filename.len() > 35 {
+        "...".to_string() + &filename[filename.len() - 32..]
```

Comment:
I think this might be what's causing https://github.com/rust-lang/rust/pull/66472#issuecomment-557693361. Maybe you could use `filename.char_indices().nth(32)` instead?